android/src/main/java/com/smileidentity/react/views/SmileIDSmartSelfieCaptureView.kt

```diff
@@ -42,22 +72,147 @@ class SmileIDSmartSelfieCaptureView(context: ReactApplicationContext) : SmileIDS
+              uiState,
+              viewModel,
+            )
+            else -> RenderSelfieCaptureScreen(userId, jobId, allowAgentMode ?: true, viewModel)
+          }
+        })
+      }
+    }
+  }
+
+  @Composable
+  private fun RenderSelfieCaptureScreen(
+    userId: String,
+    jobId: String,
+    allowAgentMode: Boolean,
+    viewModel: SelfieViewModel,
+  ) {
+    Box(
+      modifier = Modifier
+        .background(color = Color.White)
+        .windowInsetsPadding(WindowInsets.statusBars)
+        .consumeWindowInsets(WindowInsets.statusBars)
+        .fillMaxSize(),
+    ) {
+      SelfieCaptureScreen(
+        userId = userId,
+        jobId = jobId,
+        allowAgentMode = allowAgentMode,
+        allowNewEnroll = false,
+        skipApiSubmission = true,
+        viewModel = viewModel,
+      )
+    }
+  }
+
+  @Composable
+  private fun HandleSelfieConfirmation(
+    showConfirmation: Boolean,
+    uiState: SelfieUiState,
+    viewModel: SelfieViewModel,
+  ) {
+    if (showConfirmation) {
+      ImageCaptureConfirmationDialog(
+        titleText = stringResource(R.string.si_smart_selfie_confirmation_dialog_title),
+        subtitleText = stringResource(R.string.si_smart_selfie_confirmation_dialog_subtitle),
+        painter = BitmapPainter(
+          BitmapFactory
+            .decodeFile(uiState.selfieToConfirm!!.absolutePath)
+            .asImageBitmap()
+        ),
+        confirmButtonText = stringResource(R.string.si_smart_selfie_confirmation_dialog_confirm_button),
+        onConfirm = {
+          viewModel.submitJob()
+        },
+        retakeButtonText = stringResource(R.string.si_smart_selfie_confirmation_dialog_retake_button),
+        onRetake = viewModel::onSelfieRejected,
+        scaleFactor = 1.25f,
+      )
+    } else {
+      // If confirmation is disabled, automatically confirm
+      viewModel.submitJob()
+    }
+  }
+
+  @Composable
+  private fun HandleProcessingState(viewModel: SelfieViewModel) {
+    try {
+      viewModel.onFinished { res ->
+        when (res) {
+          is SmileIDResult.Success -> {
+            res.data?.let { data ->
+              val result = SmartSelfieCaptureResult(
+                selfieFile = data.selfieFile,
+                livenessFiles = data.livenessFiles ?: emptyList(),
+                apiResponse = data.apiResponse,
+              )
+              handleResultCallback(SmileIDResult.Success(SmartSelfieResult(
+                selfieFile = data.selfieFile,
+                livenessFiles = data.livenessFiles,
+                apiResponse = data.apiResponse
+              )))
+            } ?: run {
+              emitFailure(Exception("No data available in success result"))
+            }
+          }
+          is SmileIDResult.Error -> {
+            emitFailure(res.throwable ?: Exception("Unknown error occurred"))
+          }
+          else -> {
+            emitFailure(Exception("Unexpected result type"))
           }
         }
       }
```

Comment:
```suggestion
      viewModel.onFinished { res -> handleResultCallback(res)}
```
You should be able to use this function here: https://github.com/smileidentity/react-native/blob/454f213c559143ac96ed1f891a2e1efb3cc50793/android/src/main/java/com/smileidentity/react/views/SmileIDSelfieView.kt#L14